omise/setting.php

```diff
@@ -6,62 +6,82 @@
 class Setting
 {
     
-    const PREFIX = 'omise_';
+    const
+        PREFIX = 'omise_',
+        SUBMIT_ACTION = 'omise_save_setting'
+    ;
 
-    protected
-        $submit_action = 'omise_save_setting',
+    public
         $all_settings = array(
             'module_status',
             'sandbox_status',
             'test_public_key',
             'test_secret_key',
             'live_public_key',
             'live_secret_key',
-            'title',
-            'three_domain_secure_status',
-            'internet_banking_status',
-            'alipay_status'
+            'title'
+        )
+    ;
+
+    protected
+        $callToCFG = array(
+            // retrieve normal xxx_yyy_zzz config values by calling getXxxYyyZzz
+            array( 'match'=>'%^get[A-Z].*$%', 'find'=>array('%([A-Z])([a-z])%', '%get_%'), 'repl'=>array('_\1\2', '') ),
```

Comment:
It could indeed be replaced in the future. It was better at the point in time when this was implemented however to stick with `__call` so that all legacy code that was calling the `getXXXXX` functions would still work